.github/workflows/build.yaml

```diff
@@ -0,0 +1,34 @@
+name: Kubernetes-sigs/ingress2gateway build
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+permissions:
+  contents: read
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    environment: release
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.18'
```

Comment:
Any reason not to just copy the goreleaser example nearly verbatim here? https://github.com/goreleaser/goreleaser-action#workflow

Reply:
I accidentally committed the 1.20 so I revert, the reason it is `1.18` is because the i2gw code is `1.18`

> Any reason not to just copy the goreleaser example nearly verbatim here? https://github.com/goreleaser/goreleaser-action#workflow

It is nearly identical just that I am using setup-go@v3 and not v4 like in the example and I also use build and not release here.